### PR TITLE
[SW-1337] In Manual cluster mode, don't kill the cluster on unhealthy state. It is solely up on the user to decide what to to do with the application

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala [hunk 1]
```diff
@@ -244,20 +251,24 @@ class ExternalH2OBackend(val hc: H2OContext) extends SparklingBackend with Exter
...
+  }
 
+  def startUnhealthyStateCheckThread(): Unit = {
     cloudHealthCheckThread = Some(new Thread {
       override def run(): Unit = {
         while (true) {
           Thread.sleep(hc.getConf.healthCheckInterval)
           if (!H2O.CLOUD.healthy()) {
-            Log.err("External H2O cloud not healthy!!")
+            if(hc.getConf.isManualClusterStartUsed){
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala]: nit: meabe a shorter implementation?:
```
val level = if (hc.getConf.isManualClusterStartUsed) Log.WARN else Log.ERRR
Log.log(level, "External H2O cluster not healthy!")
```

Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala]: nit: missing white space after `if` and before `{`

File: doc/src/site/sphinx/configuration/configuration_properties.rst [hunk 2]
```diff
@@ -310,11 +310,13 @@ External backend configuration properties
 | ``spark.ext.h2o.external.kill.on.unhealthy``          | ``true``       | If true, the client will try to     |
 |                                                       |                | kill the cluster and then itself in |
 |                                                       |                | case some nodes in the cluster      |
-|                                                       |                | report unhealthy status.            |
+|                                                       |                | report unhealthy status. Has effect |
+|                                                       |                | only in auto cluster start mode.    |
```
Comment (mn-mikke) [inline doc/src/site/sphinx/configuration/configuration_properties.rst]: nit: auto -> the automatic?

Comment (jakubhava) [reply doc/src/site/sphinx/configuration/configuration_properties.rst]: Good catch!

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala (hunk 1 above)
Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala]: thx!

Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala]: Perfect idea!

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala [hunk 3]
```diff
@@ -229,13 +229,20 @@ class ExternalH2OBackend(val hc: H2OContext) extends SparklingBackend with Exter
       }
     }
 
-    if (hc.getConf.isKillOnUnhealthyClusterEnabled) {
+    startUnhealthyStateKillThread()
```
Comment (Tagar) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala]: Is this UnhealthyStateKillThread actualyl working .. we have an issue with 3.26 release and 
backend cluster can't start up - sent more details in an email

**Pull request merged**